tests/cupy_tests/lib_tests/test_polynomial.py

```diff
@@ -293,52 +308,65 @@ def test_poly1d_routine(self, xp, dtype):
         a1 = self._get_input(xp, self.type_l, dtype)
         a2 = self._get_input(xp, self.type_r, dtype)
         with cupyx.allow_synchronize(False):
-            out = func(a1, a2)
-        return out
+            return func(a1, a2)
+
+
+@testing.gpu
+@testing.parameterize(*testing.product({
+    'type_l': ['poly1d', 'ndarray', 'python_scalar', 'numpy_scalar'],
+    'type_r': ['poly1d', 'ndarray', 'python_scalar', 'numpy_scalar'],
+}))
+class TestPolyaddInputCombination(Poly1dTestBase):
+
+    @testing.for_dtypes((numpy.bool_,))
+    @testing.numpy_cupy_array_equal()
+    def test_polyadd(self, xp, dtype):
+        a1 = self._get_input(xp, self.type_l, dtype)
+        a2 = self._get_input(xp, self.type_r, dtype)
+        with cupyx.allow_synchronize(False):
+            return xp.polyadd(a1, a2)
 
 
 class UserDefinedArray:
 
     __array_priority__ = cupy.poly1d.__array_priority__ + 10
 
     def __init__(self):
-        self.count = 0
-        self.rcount = 0
+        self.op_count = 0
+        self.rop_count = 0
 
     def __add__(self, other):
-        self.count += 1
+        self.op_count += 1
 
     def __radd__(self, other):
-        self.rcount += 1
+        self.rop_count += 1
 
     def __sub__(self, other):
-        self.count -= 1
+        self.op_count -= 1
```

Comment:
Why did you implement to decrement `op_count`?

Reply:
No special reason, probably to match the add/sub terminologies. Is there something wrong with it? Do you prefer to always increment it to reflect the number of operations?

Reply:
Please increment `op_count` in both operations, or use `add_count` and `sub_count`.